src-java/kilda-model/src/main/java/org/openkilda/model/CookieBase.java

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.model;
+
+import org.openkilda.exception.InvalidCookieException;
+import org.openkilda.model.bitops.BitField;
+import org.openkilda.model.bitops.NumericEnumField;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@EqualsAndHashCode(of = {"value"})
+public class CookieBase implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    // update ALL_FIELDS if modify fields list
+    static final BitField TYPE_FIELD = new BitField(0x1FF0_0000_0000_0000L);
+    static final BitField SERVICE_FLAG = new BitField(0x8000_0000_0000_0000L);
+
+    // used by unit tests to check fields intersections
+    static final BitField[] ALL_FIELDS = new BitField[]{SERVICE_FLAG, TYPE_FIELD};
+
+    private final long value;
+
+    CookieBase(long value) {
+        this.value = value;
+    }
+
+    protected CookieBase(long blank, CookieType type) {
+        value = setField(blank, TYPE_FIELD, type.getValue());
+    }
+
+    /**
+     * Validate cookie value without throwing exception in case of validation fail, but returning {@code false} result.
+     */
+    public boolean safeValidate() {
```

Comment:
nit: I'd call it "isValid" and swap with "validate" to be overwritten by inheritors.

Reply:
In this case it can't report what exactly goes wrong (not this info brought with exception raised by `validate`).

PS `safeValidate` renamed into `isValid`.